app/test/WebsiteGenerator/TemplateRenderer/TwigExtension/SlidesEmbedExtensionTest.php

```diff
@@ -0,0 +1,119 @@
+            $actual
+        );
+    }
+
+    /**
+     * When the slides url is a PDF, we should get an iFrame back.
+     * The iFrame should link to the mozilla tooling which helps show the PDF.
+     */
+    public function testGetSlidesEmbedPdf()
+    {
+        $talk = new Talk();
+        $talk->setSlidesUrl('https://slides.phpsw.uk/2018/01/command-emission-promise.pdf');
+
+        $slidesEmbedExtension = new SlidesEmbedExtension();
+        $actual = $slidesEmbedExtension->getSlidesEmbed($talk);
+        $this->assertEquals(
+            '<iframe class="w-full" style="height:300px" src="https://mozilla.github.io/pdf.js/web/viewer.html?file=https://slides.phpsw.uk/2018/01/command-emission-promise.pdf#zoom=page-fit" frameborder="0" allowfullscreen mozallowfullscreen webkitallowfullscreen></iframe>',
+            $actual
+        );
+    }
+
+    /**
+     * When the slides url is slideshare, we get an embedly embed back.
+     */
+    public function testGetSlidesEmbedSlideshare()
+    {
+        $talk = new Talk();
+        $talk->setSlidesUrl('https://www.slideshare.net/asgrim1/crafting-quality-php-applications-an-overview-phpsw-march-2018');
+        $talk->setTitle('Crafting Quality PHP Applications: an overview');
+
+        $slidesEmbedExtension = new SlidesEmbedExtension();
+        $actual = $slidesEmbedExtension->getSlidesEmbed($talk);
+        $this->assertEquals(
+            '<blockquote class="embedly-card"><h4><a href="https://www.slideshare.net/asgrim1/crafting-quality-php-applications-an-overview-phpsw-march-2018">Crafting Quality PHP Applications: an overview</a></h4></blockquote><script async src="//cdn.embedly.com/widgets/platform.js" charset="UTF-8"></script>',
+            $actual
+        );
+    }
+
+    /**
+     * When the slides url is a link to a website, we should get a standard link back.
+     */
+    public function testGetSlidesLinkStandardLink()
+    {
+        $talk = new Talk();
+        $talk->setSlidesUrl('https://talks.philsturgeon.uk/instances/phpsw-jan18/http-caching-to-save-the-polar-bears/html/#/');
+
+        $slidesEmbedExtension = new SlidesEmbedExtension();
+        $actual = $slidesEmbedExtension->getSlidesLink($talk);
+        $this->assertEquals(
+            '<a href="https://talks.philsturgeon.uk/instances/phpsw-jan18/http-caching-to-save-the-polar-bears/html/#/" class="block mb-1" target="_blank">
+                <svg class="fill-current w-4 mr-1" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 20 20"><path d="M9.26 13a2 2 0 0 1 .01-2.01A3 3 0 0 0 9 5H5a3 3 0 0 0 0 6h.08a6.06 6.06 0 0 0 0 2H5A5 5 0 0 1 5 3h4a5 5 0 0 1 .26 10zm1.48-6a2 2 0 0 1-.01 2.01A3 3 0 0 0 11 15h4a3 3 0 0 0 0-6h-.08a6.06 6.06 0 0 0 0-2H15a5 5 0 0 1 0 10h-4a5 5 0 0 1-.26-10z"></path></svg>
```

Comment:
Do we want to include the markup within the texts? I've usually tried to avoid this and focus on the data properties and methods, just so that the tests don't break if a class is changed in a template.

Reply:
The problem is, the main thing we're doing with this extension is generating the relevant markup to embed the slides for a given service.

We could split each slides service up into it's own class and sort of handle it that way, which might make things a bit neater

Reply:
Maybe we keep it this way for now, and then look at refactoring it into different services post-launch?